in_toto/verifylib.go

```diff
@@ -533,19 +533,93 @@ func VerifyLayoutSignatures(layoutMb Metablock,
 	return nil
 }
 
+/*
+Merges the materials of the first step (as mentioned in the layout)
+and the products of the last step and returns a new link.
+This link reports the materials and products and summarizes the
+overall software supply chain.
+NOTE: The assumption is that the steps mentioned in the layout are
+to be performed sequentially. So, the first step mentioned in the
+layout denotes what comes into the supply chain and the last step
+denotes what goes out.
+*/
+func GetSummaryLink(layout Layout, stepsMetadataReduced map[string]Metablock) (Metablock, error) {
+	var summaryLink Link
+	var result Metablock
+	if len(layout.Steps) > 0 {
+		firstStepLink := stepsMetadataReduced[layout.Steps[0].Name]
+		lastStepLink := stepsMetadataReduced[layout.Steps[len(layout.Steps)-1].Name]
+
+		summaryLink.Materials = firstStepLink.Signed.(Link).Materials
+		/* Using the name of the first link for the summary link can be misleading
+		as it's likely not descriptive of all the steps in the sublayout. The first step
+		of the sublayout may also be rather trivial, such as extracting an archive, and
+		it could also be non-unique. The superlayout and the sublayout may share a similar
+		step which can be confusing.
+
+		It's possible to chain together all the step names into a string as follows:
+		"<step_name>, <step_name>...",
+		but this can get very long if we have nested sublayouts.
+		*/
```

Comment:
Thanks for bringing this to our attention. It is something that also needs fixing in the reference implementation. I created a ticket at https://github.com/in-toto/in-toto/issues/254. Please consider applying my proposed fix for this issue here.